server.py

```diff
@@ -208,8 +208,8 @@ def save_response(bound_logger, survey_response):
 def save_feedback_response(bound_logger, survey_feedback_response):
     bound_logger.info("Saving feedback response")
     invalid = survey_feedback_response.get("invalid")
-    survey = survey_feedback_response.get("survey_type")
-    period = survey_feedback_response.get("survey_period")
+    survey = survey_feedback_response.get("associated_survey")
+    period = survey_feedback_response.get("collection").get("period")
```

Comment:
If 'collection' is not a key you will raise an AttributeError. Is this the behaviour you want?